lib/spack/spack/environment/environment.py

```diff
@@ -576,6 +593,15 @@ def specs_for_view(self, concretized_root_specs):
 
         return specs
 
+    def use_renameat2(self):
+        if os.path.islink(self.root):
+            return False
+        elif os.path.isdir(self.root):
+            if not spack.util.atomic_update.renameat2:
+                raise Exception
+
+        return bool(spack.util.atomic_update.renameat2)
```

Comment:
I am pretty convinced that this still needs to be a config option, not an automatic decision.  Reason:

1. There are sites that share a filesystem across OS's.  They may want to ensure that views are usable across OS's with older kernels *and* OS's with newer kernels.
2. People may actually want the symlink for whatever reason.
3. This feature isn't well tested outside LLNL, and I would rather introduce it in a way that doesn't risk breaking everyone's view until we've had time to use it in production.  Putting it behind a feature flag (config option) will do that.

So we should add something like `update_method` here:

```yaml
view:
    root: /path/to/root
    update_method: [symlink | renameat2]
    link: [roots|all|run]
    # etc...    
```

And I think the default `update_method` should stay `symlink` until we can play with this more.  With the config option I'm way less worried about some unexpected consequence of this for environments that are *not* LC's (i.e., everyone else's)

Ideally, the code here would be smart enough to look at the config option and replace the view regardless of whether it was previously done with `renameat2` or a `symlink`, but if there is a config option I don't think we need to gate the merge on this, so long as we error out gracefully.